snooty/parser.py

```diff
@@ -893,15 +886,35 @@ def replace_include(node: Dict[str, Any]) -> Dict[str, Any]:
                 include_node_children = include_file_page_ast["children"]
 
                 # Resolve includes within include node
-                replaced_include = list(map(replace_include, include_node_children))
+                replaced_include = list(map(replace_nodes, include_node_children))
                 node["children"] = replaced_include
+            # Replace instances of an image's name with its full path. This allows Snooty Preview to render an image by
+            # using the location of the image on the user's local machine
+            elif node.get("name") == "figure":
+                # Obtain subset of the image's path (name)
+                argument = node["argument"][0]
```

Comment:
I'm assuming that elsewhere in the parser we've made certain that we don't ever get a figure without a first argument? Just checking.

These PR looks really fine to me -- but for some reason when I tried to run the tests locally, I got a failure.

Reply:
Yup, this is checked in [`parser.py`](https://github.com/mongodb/snooty-parser/pull/60/files#diff-84e66ec308ed95ea541bded28cfbdff1L410-L416)!
```
if argument_text is None:
                self.diagnostics.append(
                    Diagnostic.error(
                        f'"{name}" expected a path argument', util.get_line(node)
                    )
                )
                return True
```